section3/reflection.md

```diff
@@ -1,17 +1,66 @@
+    - I think the time commitment with my work schedule was more than I could have made work effectively without taking PTO. (I work 9 am - 7 pm MST, I took time off for the class sessions, but I knew that going into Mod 0. This left most of my planned study time on Friday afternoons and weekends.) After the first week, I realized that this plan was not going to work with the mid-week assignments needing 3 hours for each homework assignment **plus** note-taking and review time.  And then the Project work time using my weekend planned study time. At that point, I chose to take extra time off on Tuesday and Thursday to have some study time after 5 pm on weekdays. I think this took a lot more time needed to complete than expected while working full-time. But knowing this is good for going into the full-time program.
 
 3. What is a Hash, and how is it different from an Array?
+    - A Hash is a way to store a group of related data that you access using keys and has no assigned order. An array stores a list of things and can be referenced using its order.
 
 4. In the space below, create a Hash stored to a variable named `pet_store`.  This hash should hold an inventory of items and the number of that item that you might find at a pet store.
 
+    ```Ruby
+    pet_store = {'dog_food' => 20, 'chew_toys' => 6, 'fish' => 100}
+    ```
+
 5. Given the following `states = {"CO" => "Colorado", "IA" => "Iowa", "OK" => "Oklahoma"}`, how would you access the value `"Iowa"`?
 
+    `states["IA"]`
+
 6. With the same hash above, how would we get all the keys?  How about all the values?
 
+    `states.keys`
+
+    `states.values`
+
 7. What is another example of when we might use a hash?  In your example, why is a hash better than an array?
+    - A hash could be used when storing user profile data (eg: name, age, member since, etc.). It has multiple data points that you would want to reference. It would be better than an Array because it is how the data relates to each other that matters. Not what order that data is stored in and in a hash you can give more meaning to what pieces of data represent.
+
+8. What questions do you still have about hashes?
+  * In the ruby documentation they reference using `@auther = author` in "Hash Keys" I did not understand this section or what the `@` signs were doing. Is this adding items to the Hash, somehow?
```

Comment:
This is called an instance variable. You will use them more once you start learning about attributes.